test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java

```diff
@@ -987,7 +993,7 @@ public void directSymlink_forTestingZip() throws Exception {
             assertThat(zipPage.getWebResponse().getStatusCode(), equalTo(HttpURLConnection.HTTP_OK));
 
             List<String> entryNames = getListOfEntriesInDownloadedZip((UnexpectedPage) zipPage);
-            assertThat(entryNames, hasSize(0));
+            assertThat(entryNames, hasSize(6));
```

Comment:
Should add a `containsInAnyOrder` assertion after this to match the expected 6 entries, like the others in this file.